chapters/annotations.tex

```diff
@@ -27,6 +28,48 @@ \section{Notation for Annotation Definitions}\label{notation-for-annotation-defi
 If the annotation is declared as an \lstinline!/*evaluable*/ parameter! the corresponding modifier is further restricted to be evaluable.
 If the annotation is declared as a \lstinline!/*literal*/ constant! the corresponding modifier is further restricted to be a literal value.
 
+
+\section{Expression Evaluation Inside Annotations}\label{expression-evaluation-inside-annotations}
+
+This section describes some differences to the evaluation of expressions inside of annotations compared to normal evaluation rules outside of annotations.
+
+
+\subsection{Enumerations for Use in Annotations}\label{enumerations-for-use-in-annotations}
+
+Several annotations make use of dedicated enumeration types.
+These enumeration types do not have the full status of being built-in types as, e.g., \lstinline!StateSelect! (\cref{stateselect}).
+Instead, they are only in scope where expressions inside annotations are evaluated, shadowing any user-defined definitions with the same names.
+
+\begin{example}
+The \lstinline!smooth! attribute of a \lstinline!Polygon! can be controlled through a model parameter, but the parameter cannot use the \lstinline!Smooth! type directly:
+\begin{lstlisting}[language=modelica]
+model BezierParameter
+  parameter Smooth smooth = Smooth.Bezier; // Error: Smooth is unknown here.
+  parameter Boolean bezier = true;         // Fine.
+  annotation(Icon(graphics = {
+    Polygon(
+      smooth = if bezier then Smooth.Bezier else Smooth.None,
+      points = {{-50, -20}, {0, 30}, {50, -20}}
+    )
+  }));
+end BezierParameter;
+\end{lstlisting}
+\end{example}
+
+\begin{nonnormative}
+When evaluating the expression $\mathit{expr}$ in the model \lstinline!M!, one can imagine it being done as in
+\begin{lstlisting}[language=modelica]
+model M_annotation
+  extends M;
+  import AnnotationEnumerations.*;
+  $\mathit{T}$ result = $\mathit{expr}$;
+end M_annotation;
+\end{lstlisting}
+where \lstinline!AnnotationEnumerations! is a hidden package containing all enumeration types used in annotations, and $\mathit{T}$ is the expected type of the expression.
+(Actual implementations will also need to avoid potential shadowing problems involving the names \lstinline!M_annotation! and \lstinline!result!.)
```

Comment:
Talking about possible implementation issues with this approach makes it seem like this is suggestion on how to implement it, which I assume isn't really the purpose. So I would suggest just removing this sentence.

I'm not sure if the whole nonnormative section is really needed at all actually, annotations aren't really different to normal expressions except in regard to what built-in types are available.

Reply:
I am open to removing the sentence.

The intention of the non-normative section was precisely to show that the evaluation of expressions actually is very similar to evaluation elsewhere, by showing how the expression can be placed in a hypothetical non-annotation context for evaluation.

Reply:
Only removing the last sentence but leaving the rest would be fine with me.

Reply:
Done.  Resolving.